overlord/snapstate/snapstate.go

```diff
@@ -2045,40 +1813,89 @@ type UpdateTaskSets struct {
 	Refresh []*state.TaskSet
 }
 
-func doUpdate(ctx context.Context, st *state.State, names []string, updates []minimalInstallInfo, params updateParamsFunc, userID int, globalFlags *Flags, prqt PrereqTracker, deviceCtx DeviceContext, fromChange string) ([]string, *UpdateTaskSets, error) {
-	if globalFlags == nil {
-		globalFlags = &Flags{}
+type update struct {
+	Setup      SnapSetup
+	SnapState  SnapState
+	Components []ComponentSetup
```

Comment:
These can be unexported

Reply:
These fields are used in some tests that re-export functions and some types, so they have to stay exported unfortunately.

Reply:
Hmm, I see... it would be possible to do some type translation in export_test.go but I guess it is not worth it.